### DISCUSSION TITLE
Too many queries using getUrl() method

### DISCUSSION THREAD

**Discussion by darkons** [spatie/laravel-medialibrary#2838] [Q&A] [answered]
When you call Media getUrl() method, it's result in a new model query for each media item.
Model to reproduce issue:

```
class Media extends BaseMedia
{
    protected $appends = [
        'url'
    ];

    public function getUrlAttribute()
    {
        return $this->getUrl();
    }
}
```

Execute the query:
```
$c = Customer::with('media')->find(1);
^ "mysql (0.74): select * from `customers` where `customers`.`id` = 1"
^ "mysql (0.93): select * from `media` where `media`.`model_id` in (1) and `media`.`model_type` = Customer"
^ "mysql (0.6): select * from `customers` where `customers`.`id` = 1"
^ "mysql (0.58): select * from `customers` where `customers`.`id` = 1"
^ "mysql (0.83): select * from `customers` where `customers`.`id` = 1"
..........
```
In a collection of 100 objects it will execute 101 queries

Could this issue be solved somehow?

**Comment by erikn69** [comment, marked as answer]
For that it exists eager loading on laravel, you only have to read documentation

**Reply by darkons** [reply]
Yeah I know and I'm using eager loading on Customer::**with('media')** but every time getUrl() is called on each Media item, a new query is executed over customers table with same ID.
If I removed "url" from model appends then only the first two queries are executed.

How am I supposed to use eager loading in this case? My mind is broken right now :/

**Reply by erikn69** [reply]
Hard to say without laravel version, package version, and more

**Reply by darkons** [reply]
Thank you very much for your help

Laravel version: 7.30.4
Spatie Media Library version: 9.8.0
PHP version: 7.4

If you need more info, please just tell me.

Thank you!

**Reply by erikn69** [reply]
>```php
>class Media extends BaseMedia
>{
>    protected $appends = [
>        'url'
>    ];
>
>    public function getUrlAttribute()
>    {
>        return $this->getUrl();
>    }
>}
>```

I did test that, everything is working as expected, the problem is on your implementation, make a debug

**Reply by darkons** [reply]
Sorry for that. You are right!
The problem was in my CustomPathGenerator using Media relationships without eager loading.
Fixing...

Thank you for help! :+1:
